x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/index.test.tsx

```diff
@@ -746,6 +774,122 @@ describe('when on the endpoint list page', () => {
       expect(renderResult.getByTestId('endpointDetailsActionsButton')).not.toBeNull();
     });
 
+    describe('when showing Activity Log panel', () => {
+      let renderResult: ReturnType<typeof render>;
+      const agentId = 'some_agent_id';
+
+      let getMockData: () => ActivityLog;
+      beforeEach(async () => {
+        global.IntersectionObserver = jest.fn(() => ({
+          root: null,
+          rootMargin: '',
+          thresholds: [],
+          takeRecords: jest.fn(),
+          observe: jest.fn(),
+          unobserve: jest.fn(),
+          disconnect: jest.fn(),
+        }));
+
+        const fleetActionGenerator = new FleetActionGenerator(Math.random().toString());
+        const responseData = fleetActionGenerator.generateResponse({
+          agent_id: agentId,
+        });
+        const actionData = fleetActionGenerator.generate({
+          agents: [agentId],
+        });
+        getMockData = () => ({
+          page: 1,
+          pageSize: 50,
+          data: [
+            {
+              type: 'response',
+              item: {
+                id: 'some_id_0',
+                data: responseData,
+              },
+            },
+            {
+              type: 'action',
+              item: {
+                id: 'some_id_1',
+                data: actionData,
+              },
+            },
+          ],
+        });
+
+        renderResult = render();
+        await reactTestingLibrary.act(async () => {
+          await middlewareSpy.waitForAction('serverReturnedEndpointList');
+        });
+        const hostNameLinks = await renderResult.findAllByTestId('hostnameCellLink');
+        if (hostNameLinks.length) {
+          reactTestingLibrary.fireEvent.click(hostNameLinks[0]);
```

Comment:
surprised you did not have to `act()` this click event.

Also, I see this wrapped in an `if` statement. should the tests fail if there are no hostNameLinks? if so, I would instead change the `getAllByTestId()` - because that would `throw` if there are none found

Reply:
done 6f34ef606c94b565e2a1bf8ef7a6077cb645744d